src/edge/simulator.ts

```diff
@@ -38,16 +41,20 @@ export class Simulator {
             } else {
                 message = Constants.updateSimulatorMsg;
             }
+            type = "upgrade";
         } catch (error) {
             message =  Constants.needSimulatorInstalledMsg;
+            type = "install";
         }
 
         if (message) {
```

Comment:
Need to check type value here
```suggestion
        if (message) {
```

Reply:
type is the telemetry name. message has value means we need to install or update the iotedgehubdev tool. I will add default value to type as empty string to be safer.

Reply:
code updated